README.md
[![Build Status](https://gitlab.gnome.org/GNOME/gnome-software/badges/master/build.svg)](https://gitlab.gnome.org/GNOME/gnome-software/pipelines)

# Software

[Software](https://wiki.gnome.org/Apps/Software) lets you install and update applications and system extensions.
A plugin system is used to access software from different sources.
Plugins are provided for:
 - Traditional package installation via PackageKit (e.g. Debian package, RPM).
 - Next generation packages: [Flatpak](https://flatpak.org/) and [Snap](https://snapcraft.io/).
 - Firmware updates.
 - Ratings and reviews ([ODRS](https://odrs.gnome.org/) and Ubuntu reviews).
 - [Steam](http://store.steampowered.com/) games.

Software supports showing metadata that closely matches the [AppStream](https://www.freedesktop.org/wiki/Distributions/AppStream/) format.

Software runs as a background service to provide update notifications and be a search provider for [GNOME Shell](https://wiki.gnome.org/Projects/GnomeShell).

# Building

Build locally with:
```
$ meson --prefix $PWD/install build/
$ ninja -C build/ all install
$ killall gooroom-software
$ XDG_DATA_DIRS=install/share:$XDG_DATA_DIRS ./install/bin/gooroom-software
```

# Debugging

Running with `--verbose` will give detailed logging information.


debian/changelog
gooroom-software (3.30.6+5+grm3u7) unstable; urgency=medium

  [ heejun.jang ]
  * [cc94b14b] Update gooroom software to apply interface text size change and resize category item size

 -- Gooroom Autobuilder <jenkins@gooroom.kr>  Mon, 13 Nov 2023 16:27:57 +0900

gooroom-software (3.30.6+5+grm3u6) unstable; urgency=medium

  [ heejun.jang ]
  * [e95f0e40] Update update-button show desc action, change window to scrolledwindow, resize category item size
  * [052cf4b2] Update gooroom software to apply interface text change

 -- Gooroom Autobuilder <jenkins@gooroom.kr>  Fri, 28 Jul 2023 09:46:35 +0900

gooroom-software (3.30.6+5+grm3u5) unstable; urgency=medium

  [ heejun.jang ]
  * [ed68889e] update os info

 -- Gooroom Autobuilder <jenkins@gooroom.kr>  Fri, 30 Jun 2023 19:44:18 +0900

gooroom-software (3.30.6+5+grm3u4) unstable; urgency=medium

  [ heejun.jang ]
  * [1023f8ba] [HGOOROOM-180]Updated similar more button do not show when do not have category

 -- Gooroom Autobuilder <jenkins@gooroom.kr>  Mon, 20 Jun 2022 08:47:26 +0900

gooroom-software (3.30.6+5+grm3u3) unstable; urgency=medium

  [ heejun.jang ]
  * [50e424cd] [Bug:5095]Enable the help program

 -- Gooroom Autobuilder <jenkins@gooroom.kr>  Thu, 26 May 2022 15:47:19 +0900

gooroom-software (3.30.6+5+grm3u2) unstable; urgency=medium

  * [1c1fb2b1] Added group package for Gooroom-3.0

 -- Gooroom Autobuilder <jenkins@gooroom.kr>  Sun, 27 Feb 2022 15:40:11 +0900

gooroom-software (3.30.6+5+grm3u1) unstable; urgency=medium

  [ donghun ]
  * [3e3ece0a] Added user consent installation and group installation function
  * [e368e98b] [Bug:2583,2598,2599,2902,3095,3096]applied gooroom-3.0 and fix errors
  * [ae257f7e] Rebuild gooroom-2.0

  [ heejun.jang ]
  * [ad75922f] Rebase to Gooroom-3.0
  * [c7f767e0] Updated catebory menu item background opacity
  * [b4b91b68] [Bug:3965]Deleted empty space on category
  * [a07f91b5] [Bug:3980]Update OpenSCAD application id to recent version
  * [4fc17043] [Bug:4167]Fixed to do not show the duplicated applications
  * [125dba55] [Bug:4296]Fixed to do not show the installed applications which do not have desktop file info in search page
  * [e2eb6160] Fixed StaticAnalysis Severity High Issue - Uninitialized Variable

 -- Gooroom Autobuilder <jenkins@gooroom.kr>  Sat, 04 Dec 2021 17:20:23 +0900

gooroom-software (3.30.6+5+grm2u6) unstable; urgency=medium

  [ donghun ]
  * [3baa8937] Added installation list updates after installation is complete

  [ Gooroom Autobuilder ]

 -- Gooroom Autobuilder <jenkins@gooroom.kr>  Sat, 11 Jul 2020 08:57:39 +0900

gooroom-software (3.30.6+5+grm2u5) unstable; urgency=medium

  [ donghun ]
  * [2cc9a035] Modified the compatible programs

  [ Gooroom Autobuilder ]

 -- Gooroom Autobuilder <jenkins@gooroom.kr>  Wed, 08 Jul 2020 20:28:09 +0900

gooroom-software (3.30.6+5+grm2u4) unstable; urgency=medium

  [ donghun ]
  * [16c19552] [Bug:2029, 2269] Updated the sorting of the install information in the installed page and added spinner in the category page

  [ Gooroom Autobuilder ]

 -- Gooroom Autobuilder <jenkins@gooroom.kr>  Tue, 07 Jul 2020 14:23:55 +0900

gooroom-software (3.30.6+5+grm2u3) unstable; urgency=medium

  [ donghun ]
  * [1307c9bb] Fixed segmentation fault error

  [ Gooroom Autobuilder ]

 -- Gooroom Autobuilder <jenkins@gooroom.kr>  Mon, 17 Feb 2020 10:04:03 +0900

gooroom-software (3.30.6+5+grm2u2) unstable; urgency=medium

  [ donghun ]
  * [3e6f85b2] Update user information location for GPMS linkage
  * [bc4edacf] Added repository usage settings (Debian Packages,Flathub,Gooroom)
  * [b9af1202] Added desktop file delete when deleting flatpak application
  * [76b209a9] Added default repository (flathub)

  [ Gooroom Autobuilder ]

 -- Gooroom Autobuilder <jenkins@gooroom.kr>  Tue, 11 Feb 2020 11:29:09 +0900

gooroom-software (3.30.6+5+grm2u1) unstable; urgency=medium

  [ donghun ]
  * [748e3221] Auto start remove and detail page update
  * [60ac259e] Modified to search less than 2 words
  * [f564e0f6] Fixed category popup activation bug
  * [c05490c6] Modify software center server schema and fix installation dependencies
  * [275f0a9a] Fixed other category exposure error
  * [f6036f87] Fixed category name error
  * [e2c462cb] Fixed popular apps and featured apps exposure error
  * [96fe0235] Delete search text when moving menu
  * [6a0c9c2f] Fixed back button error
  * [b5d3ebde] Fixed app update error
  * [f2bb2d35] Modified update UI
  * [02268925] Fixed popular app list and modified recent app sort
  * [49c6a9f5] Update source regarding copyright
  * [4622e373] Fixed button status error when canceling during Debian app installation
  * [afbc871b] Added setting of appstream activation
  * [9b9ec129] Fixed search error and detail UI

  [ Gooroom Autobuilder ]

 -- Gooroom Autobuilder <jenkins@gooroom.kr>  Sun, 01 Dec 2019 19:35:37 +0900

gooroom-software (3.30.6+5) unstable; urgency=medium

  * Initial release

 -- Gooroom <gooroom@gooroom.kr>  Tue, 05 Jun 2019 12:00:00 -0500

gnome-software (3.30.6-5) unstable; urgency=medium

  * Cherry-pick 7 Snap patches via Ubuntu
    - Support switching channels, add verified developer support, use
      "wide scope" searching (search all channels), and use the new media API
  * Bump minimum libsnapd-glib-dev to 1.45

 -- Jeremy Bicha <jbicha@debian.org>  Sun, 24 Feb 2019 21:03:00 -0500

gnome-software (3.30.6-4) unstable; urgency=medium

  * Re-enable Flatpak on mipsel
  * Don't enable snap on ppc64 since snapd never built there

 -- Jeremy Bicha <jbicha@debian.org>  Thu, 10 Jan 2019 17:25:24 -0500

gnome-software (3.30.6-3) unstable; urgency=medium

  * Don't try building Flatpak support on architectures where Flatpak
    isn't built

 -- Jeremy Bicha <jbicha@debian.org>  Sun, 06 Jan 2019 01:44:44 -0500

gnome-software (3.30.6-2) unstable; urgency=medium

  * Recommend instead of Suggest fwupd (Closes: #916036)
  * Enable fwupd on all Linux architectures
  * Drop obsolete Build-Depends on intltool

lib/gs-app.h
/* -*- Mode: C; tab-width: 8; indent-tabs-mode: t; c-basic-offset: 8 -*-
 *
 * Copyright (C) 2013-2018 Richard Hughes <richard@hughsie.com>
 * Copyright (C) 2014-2015 Kalev Lember <klember@redhat.com>
 * Copyright (C) 2018-2019 Gooroom <gooroom@gooroom.kr>
 *
 * Licensed under the GNU General Public License Version 2
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */

#ifndef __GS_APP_H
#define __GS_APP_H

#include <glib-object.h>
#include <gdk/gdk.h>
#include <gdk-pixbuf/gdk-pixbuf.h>
#include <appstream-glib.h>

#include "gs-channel.h"
#include "gs-price.h"

G_BEGIN_DECLS

#define GS_TYPE_APP (gs_app_get_type ())

G_DECLARE_DERIVABLE_TYPE (GsApp, gs_app, GS, APP, GObject)

struct _GsAppClass
{
	GObjectClass		 parent_class;
	void			 (*to_string)	(GsApp		*app,
						 GString	*str);
	gpointer		 padding[30];
};

/**
 * GsAppKudo:
 * @GS_APP_KUDO_MY_LANGUAGE:		Localised in my language
 * @GS_APP_KUDO_RECENT_RELEASE:		Released recently
 * @GS_APP_KUDO_FEATURED_RECOMMENDED:	Chosen for the front page
 * @GS_APP_KUDO_MODERN_TOOLKIT:		Uses a modern toolkit
 * @GS_APP_KUDO_SEARCH_PROVIDER:	Provides a search provider
 * @GS_APP_KUDO_INSTALLS_USER_DOCS:	Installs user docs
 * @GS_APP_KUDO_USES_NOTIFICATIONS:	Registers notifications
 * @GS_APP_KUDO_HAS_KEYWORDS:		Has at least 1 keyword
 * @GS_APP_KUDO_USES_APP_MENU:		Uses an AppMenu for navigation
 * @GS_APP_KUDO_HAS_SCREENSHOTS:	Supplies screenshots
 * @GS_APP_KUDO_POPULAR:		Is popular
 * @GS_APP_KUDO_PERFECT_SCREENSHOTS:	Supplies perfect screenshots
 * @GS_APP_KUDO_HIGH_CONTRAST:		Installs a high contrast icon
 * @GS_APP_KUDO_HI_DPI_ICON:		Installs a HiDPI icon
 * @GS_APP_KUDO_SANDBOXED:		Application is sandboxed
 * @GS_APP_KUDO_SANDBOXED_SECURE:	Application is sandboxed securely
 * @GS_APP_KUDO_FEATURED:           Is featured in gooroom
 *
 * Any awards given to the application.
 **/
typedef enum {
	GS_APP_KUDO_MY_LANGUAGE			= 1 << 0,
	GS_APP_KUDO_RECENT_RELEASE		= 1 << 1,
	GS_APP_KUDO_FEATURED_RECOMMENDED	= 1 << 2,
	GS_APP_KUDO_MODERN_TOOLKIT		= 1 << 3,
	GS_APP_KUDO_SEARCH_PROVIDER		= 1 << 4,
	GS_APP_KUDO_INSTALLS_USER_DOCS		= 1 << 5,
	GS_APP_KUDO_USES_NOTIFICATIONS		= 1 << 6,
	GS_APP_KUDO_HAS_KEYWORDS		= 1 << 7,
	GS_APP_KUDO_USES_APP_MENU		= 1 << 8,
	GS_APP_KUDO_HAS_SCREENSHOTS		= 1 << 9,
	GS_APP_KUDO_POPULAR			= 1 << 10,
	GS_APP_KUDO_PERFECT_SCREENSHOTS		= 1 << 12,
	GS_APP_KUDO_HIGH_CONTRAST		= 1 << 13,
	GS_APP_KUDO_HI_DPI_ICON			= 1 << 14,
	GS_APP_KUDO_SANDBOXED			= 1 << 15,
	GS_APP_KUDO_SANDBOXED_SECURE		= 1 << 16,
	GS_APP_KUDO_FEATURED		    = 1 << 17,
	/*< private >*/
	GS_APP_KUDO_LAST
} GsAppKudo;

#define	GS_APP_INSTALL_DATE_UNSET		0
#define	GS_APP_INSTALL_DATE_UNKNOWN		1 /* 1s past the epoch */
#define	GS_APP_SIZE_UNKNOWABLE			G_MAXUINT64

/**
 * GsAppQuality:
 * @GS_APP_QUALITY_UNKNOWN:	The quality value is unknown
 * @GS_APP_QUALITY_LOWEST:	Lowest quality
 * @GS_APP_QUALITY_NORMAL:	Normal quality
 * @GS_APP_QUALITY_HIGHEST:	Highest quality
 *
 * Any awards given to the application.
 **/
typedef enum {
	GS_APP_QUALITY_UNKNOWN,
	GS_APP_QUALITY_LOWEST,
	GS_APP_QUALITY_NORMAL,
	GS_APP_QUALITY_HIGHEST,
	/*< private >*/
	GS_APP_QUALITY_LAST
} GsAppQuality;

GsApp		*gs_app_new			(const gchar	*id);
G_DEPRECATED_FOR(gs_app_set_from_unique_id)
GsApp		*gs_app_new_from_unique_id	(const gchar	*unique_id);
void		 gs_app_set_from_unique_id	(GsApp		*app,
						 const gchar	*unique_id);
gchar		*gs_app_to_string		(GsApp		*app);
void		 gs_app_to_string_append	(GsApp		*app,
						 GString	*str);

const gchar	*gs_app_get_id			(GsApp		*app);
void		 gs_app_set_id			(GsApp		*app,
						 const gchar	*id);
AsAppKind	 gs_app_get_kind		(GsApp		*app);
void		 gs_app_set_kind		(GsApp		*app,
						 AsAppKind	 kind);
AsAppState	 gs_app_get_state		(GsApp		*app);
void		 gs_app_set_state		(GsApp		*app,
						 AsAppState	 state);
AsAppScope	 gs_app_get_scope		(GsApp		*app);
void		 gs_app_set_scope		(GsApp		*app,
						 AsAppScope	 scope);
AsBundleKind	 gs_app_get_bundle_kind		(GsApp		*app);
void		 gs_app_set_bundle_kind		(GsApp		*app,
						 AsBundleKind	 bundle_kind);
void		 gs_app_set_state_recover	(GsApp		*app);
guint		 gs_app_get_progress		(GsApp		*app);
void		 gs_app_set_progress		(GsApp		*app,
						 guint		 percentage);
gboolean	gs_app_get_allow_cancel		(GsApp	*app);
void		gs_app_set_allow_cancel		 (GsApp	*app,
						  gboolean	allow_cancel);
const gchar	*gs_app_get_unique_id		(GsApp		*app);
const gchar	*gs_app_get_branch		(GsApp		*app);
void		 gs_app_set_branch		(GsApp		*app,
						 const gchar	*branch);
const gchar	*gs_app_get_name		(GsApp		*app);
void		 gs_app_set_name		(GsApp		*app,
						 GsAppQuality	 quality,
						 const gchar	*name);
const gchar	*gs_app_get_source_default	(GsApp		*app);
void		 gs_app_add_source		(GsApp		*app,
						 const gchar	*source);
GPtrArray	*gs_app_get_sources		(GsApp		*app);
void		 gs_app_set_sources		(GsApp		*app,
						 GPtrArray	*sources);
const gchar	*gs_app_get_source_id_default	(GsApp		*app);
void		 gs_app_add_source_id		(GsApp		*app,
						 const gchar	*source_id);
GPtrArray	*gs_app_get_source_ids		(GsApp		*app);
void		 gs_app_set_source_ids		(GsApp		*app,
						 GPtrArray	*source_ids);
void		 gs_app_clear_source_ids	(GsApp		*app);
const gchar	*gs_app_get_project_group	(GsApp		*app);
void		 gs_app_set_project_group	(GsApp		*app,
						 const gchar	*project_group);
const gchar	*gs_app_get_desktop_group	(GsApp		*app);
void		 gs_app_set_desktop_group	(GsApp		*app,
						 const gchar	*desktop_group);
const gchar	*gs_app_get_developer_name	(GsApp		*app);
void		 gs_app_set_developer_name	(GsApp		*app,
						 const gchar	*developer_name);
const gchar	*gs_app_get_agreement		(GsApp		*app);
void		 gs_app_set_agreement		(GsApp		*app,
						 const gchar	*agreement);
const gchar	*gs_app_get_version		(GsApp		*app);
const gchar	*gs_app_get_version_ui		(GsApp		*app);
void		 gs_app_set_version		(GsApp		*app,
						 const gchar	*version);
const gchar	*gs_app_get_summary		(GsApp		*app);
void		 gs_app_set_summary		(GsApp		*app,
						 GsAppQuality	 quality,
						 const gchar	*summary);
const gchar	*gs_app_get_summary_missing	(GsApp		*app);
void		 gs_app_set_summary_missing	(GsApp		*app,
						 const gchar	*summary_missing);
const gchar	*gs_app_get_description		(GsApp		*app);
void		 gs_app_set_description		(GsApp		*app,
						 GsAppQuality	 quality,
						 const gchar	*description);
const gchar	*gs_app_get_url			(GsApp		*app,
						 AsUrlKind	 kind);
void		 gs_app_set_url			(GsApp		*app,
						 AsUrlKind	 kind,
						 const gchar	*url);
const gchar	*gs_app_get_launchable		(GsApp		*app,
						 AsLaunchableKind kind);
void		 gs_app_set_launchable		(GsApp		*app,
						 AsLaunchableKind kind,
						 const gchar	*launchable);
const gchar	*gs_app_get_license		(GsApp		*app);
gboolean	 gs_app_get_license_is_free	(GsApp		*app);
void		 gs_app_set_license		(GsApp		*app,
						 GsAppQuality	 quality,
						 const gchar	*license);
gchar		**gs_app_get_menu_path		(GsApp		*app);
void		 gs_app_set_menu_path		(GsApp		*app,
						 gchar		**menu_path);
const gchar	*gs_app_get_origin		(GsApp		*app);
void		 gs_app_set_origin		(GsApp		*app,
						 const gchar	*origin);
const gchar	*gs_app_get_origin_appstream	(GsApp		*app);
void		 gs_app_set_origin_appstream	(GsApp		*app,
						 const gchar	*origin_appstream);
const gchar	*gs_app_get_origin_hostname	(GsApp		*app);
void		 gs_app_set_origin_hostname	(GsApp		*app,
						 const gchar	*origin_hostname);
GPtrArray	*gs_app_get_screenshots		(GsApp		*app);
void		 gs_app_add_screenshot		(GsApp		*app,
						 AsScreenshot	*screenshot);
const gchar	*gs_app_get_update_version	(GsApp		*app);
const gchar	*gs_app_get_update_version_ui	(GsApp		*app);
void		 gs_app_set_update_version	(GsApp		*app,
						 const gchar	*update_version);
const gchar	*gs_app_get_update_details	(GsApp		*app);
void		 gs_app_set_update_details	(GsApp		*app,
						 const gchar	*update_details);
AsUrgencyKind	 gs_app_get_update_urgency	(GsApp		*app);
void		 gs_app_set_update_urgency	(GsApp		*app,
						 AsUrgencyKind	 update_urgency);
const gchar	*gs_app_get_management_plugin	(GsApp		*app);
void		 gs_app_set_management_plugin	(GsApp		*app,
						 const gchar	*management_plugin);
GdkPixbuf	*gs_app_get_pixbuf		(GsApp		*app);
void		 gs_app_set_pixbuf		(GsApp		*app,
						 GdkPixbuf	*pixbuf);
GsPrice		*gs_app_get_price		(GsApp		*app);
void		 gs_app_set_price		(GsApp		*app,
						 gdouble	 amount,
						 const gchar	*currency);
GPtrArray	*gs_app_get_icons		(GsApp		*app);
void		 gs_app_add_icon		(GsApp		*app,
						 AsIcon		*icon);
GFile		*gs_app_get_local_file		(GsApp		*app);
void		 gs_app_set_local_file		(GsApp		*app,
						 GFile		*local_file);
AsContentRating	*gs_app_get_content_rating	(GsApp		*app);
void		 gs_app_set_content_rating	(GsApp		*app,
						 AsContentRating *content_rating);
GsApp		*gs_app_get_runtime		(GsApp		*app);
void		 gs_app_set_runtime		(GsApp		*app,
						 GsApp		*runtime);
const gchar	*gs_app_get_metadata_item	(GsApp		*app,
						 const gchar	*key);
GVariant	*gs_app_get_metadata_variant	(GsApp		*app,
						 const gchar	*key);
void		 gs_app_set_metadata		(GsApp		*app,
						 const gchar	*key,
						 const gchar	*value);
void		 gs_app_set_metadata_variant	(GsApp		*app,
						 const gchar	*key,
						 GVariant	*value);
gint		 gs_app_get_rating		(GsApp		*app);
void		 gs_app_set_rating		(GsApp		*app,
						 gint		 rating);
GArray		*gs_app_get_review_ratings	(GsApp		*app);
void		 gs_app_set_review_ratings	(GsApp		*app,
						 GArray		*review_ratings);
GPtrArray	*gs_app_get_reviews		(GsApp		*app);
void		 gs_app_add_review		(GsApp		*app,
						 AsReview	*review);
void		 gs_app_remove_review		(GsApp		*app,
						 AsReview	*review);
GPtrArray	*gs_app_get_provides		(GsApp		*app);
void		 gs_app_add_provide		(GsApp		*app,
						 AsProvide	*provide);
guint64		 gs_app_get_size_installed	(GsApp		*app);
void		 gs_app_set_size_installed	(GsApp		*app,
						 guint64	 size_installed);
guint64		 gs_app_get_size_download	(GsApp		*app);
void		 gs_app_set_size_download	(GsApp		*app,
						 guint64	 size_download);
void		 gs_app_add_related		(GsApp		*app,
						 GsApp		*app2);
void		 gs_app_add_addon		(GsApp		*app,
						 GsApp		*addon);
void		 gs_app_add_history		(GsApp		*app,
						 GsApp		*app2);
guint64		 gs_app_get_install_date	(GsApp		*app);
void		 gs_app_set_install_date	(GsApp		*app,
						 guint64	 install_date);
GPtrArray	*gs_app_get_categories		(GsApp		*app);
void		 gs_app_set_categories		(GsApp		*app,
						 GPtrArray	*categories);
GPtrArray	*gs_app_get_key_colors		(GsApp		*app);
void		 gs_app_set_key_colors		(GsApp		*app,
						 GPtrArray	*key_colors);
void		 gs_app_add_key_color		(GsApp		*app,
						 GdkRGBA	*key_color);
gboolean	 gs_app_has_category		(GsApp		*app,
						 const gchar	*category);
void		 gs_app_add_category		(GsApp		*app,
						 const gchar	*category);
gboolean	 gs_app_remove_category		(GsApp		*app,
						 const gchar	*category);
GPtrArray	*gs_app_get_keywords		(GsApp		*app);
void		 gs_app_set_keywords		(GsApp		*app,
						 GPtrArray	*keywords);
void		 gs_app_add_kudo		(GsApp		*app,
						 GsAppKudo	 kudo);
void		 gs_app_remove_kudo		(GsApp		*app,
						 GsAppKudo	 kudo);
gboolean	 gs_app_has_kudo		(GsApp		*app,
						 GsAppKudo	 kudo);
guint64		 gs_app_get_kudos		(GsApp		*app);
guint		 gs_app_get_kudos_percentage	(GsApp		*app);
gboolean	 gs_app_get_to_be_installed	(GsApp		*app);
void		 gs_app_set_to_be_installed	(GsApp		*app,
						 gboolean	 to_be_installed);
void		 gs_app_set_match_value		(GsApp		*app,
						 guint		 match_value);
guint		 gs_app_get_match_value		(GsApp		*app);

gboolean	 gs_app_has_quirk		(GsApp		*app,
						 AsAppQuirk	 quirk);
void		 gs_app_add_quirk		(GsApp		*app,
						 AsAppQuirk	 quirk);
void		 gs_app_remove_quirk		(GsApp		*app,
						 AsAppQuirk	 quirk);
gboolean	 gs_app_is_installed		(GsApp		*app);
gboolean	 gs_app_is_updatable		(GsApp		*app);
GPtrArray	*gs_app_get_channels		(GsApp		*app);
void		 gs_app_add_channel		(GsApp		*app,
						 GsChannel	*channel);
void		 gs_app_set_active_channel	(GsApp		*app,
						 GsChannel	*channel);
GsChannel	*gs_app_get_active_channel	(GsApp		*app);
G_END_DECLS

#endif /* __GS_APP_H */

/* vim: set noexpandtab: */
